### PR TITLE
WIP libnetwork - Add support for /networks and /services API endpoints

### PR DESCRIPTION
**Pull request by abronan**
Tested with docker experimental.

The API is going to change (with the `sandbox` entity: see docker/libnetwork#365), but I open the PR to keep track of the evolution on both sides.

*TODO* items:
- [ ] Add error scheme for all methods
- [ ] Move the networks and services endpoints in their own file (`networks.go`/`services.go`?)
- [ ] Include new `sandbox` entity
- [ ] Refactor and split tests

Signed-off-by: Alexandre Beslic <abronan@docker.com>

### CODE REVIEW

**Comment by abronan** [comment]
Also, it's now probably time to cleanup the client a little bit... And split in multiple files (`containers.go`/`images.go`/`networks.go`/`services.go`, etc.)

**Comment by mavenugo** [comment]
@aboch can you please take a look ?

**Comment by chanwit** [comment]
:+1:

**Comment by jimmyxian** [comment]
Cool, split in multiple files is pretty good.

**Comment by aluzzardi** [comment]
Awesome :)

But ... aren't we going to blindly proxy the calls to the engine anyway ? :)

**Comment by jimmyxian** [comment]
@aluzzardi 
In Swarm, I think proxy the calls to the engine is a good way to keep api fresh.

File: dockerclient.go [hunk 1]
```diff
@@ -738,3 +738,242 @@ func (client *DockerClient) BuildImage(image *BuildImage) (io.ReadCloser, error)
...
 	return client.doStreamRequest("POST", uri, image.Context, headers)
 }
+
+// Networks
+
+func (client *DockerClient) ListNetworks(name string, partialId string) ([]Network, error) {
+	uri := fmt.Sprintf("/%s/networks", APIVersion)
+
+	if name != "" {
+		uri += "&name=" + name
```
Comment (aboch) [inline dockerclient.go]: We support only one query parameter for these endpoints in libentworks, either name or partial-id
I think this should be: `uri += "?name=" +name` and `uri += "?partial-id=" + partialId` below.

File: dockerclient.go [hunk 2]
```diff
@@ -738,3 +738,242 @@ func (client *DockerClient) BuildImage(image *BuildImage) (io.ReadCloser, error)
...
+	return err
+}
+
+// Services
+
+func (client *DockerClient) ListServices(name string, partialId string, network string) ([]Service, error) {
+	uri := fmt.Sprintf("/%s/services", APIVersion)
+
+	if name != "" {
+		uri += "&name=" + name
```
Comment (aboch) [inline dockerclient.go]: Same here and below, please replace the `&` with `?`

**Comment by abronan** [comment]
@aluzzardi Most of them yes ;)

@aboch Thanks! Will revisit and fix when your PR will be merged on libnetwork's side.

**Comment by abronan** [comment]
replaced by #169 that supports the new networking UX

**Pull request closed without merging**